### PR TITLE
Console pager

### PR DESCRIPTION
**Pull request by afshin**
- [x] Change `ConsolePanel` to a `SplitPanel` so different types of code introspection can reside in tabs instead of being obtrusive.
- [x] Create a `ConsoleInspector` widget to reside inside the side tabs of the console panel.
- [x] Allow multiple types of inspectors to exist with priority ranks for their automatic display as the code inside a console is inspected.
- [x] Support history functionality for console inspectors to allow users to navigate back and forth and see introspection results they've already requested.
- [x] Allow toggling of the inspector to either the right side or the bottom of the console. This is available both as commands in the palette and as buttons in the toolbar.

cf. https://github.com/jupyter/jupyterlab/issues/76

### CODE REVIEW

File: src/console/panel.ts [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+
+/**
+ * The class name added to console panels.
+ */
+const CONSOLE_PANEL = 'jp-ConsolePanel';
+
+
+/**
+ * A panel which contains a toolbar and a console.
```
Comment (blink1073) [inline src/console/panel.ts]: Toolbar?

File: src/console/panel.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+/**
+ * A namespace for ConsolePanel statics.
+ */
+export
+namespace ConsolePanel {
+  /**
+   * The initialization options for a console panel.
+   */
+  export
+    interface IOptions {
```
Comment (blink1073) [inline src/console/panel.ts]: Alignment.

Comment (afshin) [inline src/console/panel.ts]: Cheers, hawk eye.

File: src/console/panel.ts (hunk 1 above)
Comment (afshin) [inline src/console/panel.ts]: Yeah, that was the original plan and never happened. I only just updated that comment.

**Comment by blink1073** [comment]
Looks great, @afshin!  I sent you a PR with some minor cleanup.

**Comment by blink1073** [comment]
Let's take it for a test-drive...

**Comment by jasongrout** [comment]
Thanks @afshin!

**Comment by afshin** [comment]
cf. https://github.com/jupyter/jupyterlab/issues/430

**Pull request merged**